src/model/residual_predictor.py

```diff
@@ -7,28 +7,35 @@
 
 class ResidualPredictor(nn.Module):
     def __init__(self, input_size: int = 128, hidden_size: int = 32, output_size: int = 120, num_layers: int = 2,
-                 hidden_dim: int = 120, nhead: int = 2, time_steps: int = 5, log: bool = True) -> None:
+                 hidden_dim: int = 120, embed_dim: int = 8, nhead: int = 2, time_steps: int = 5, log: bool = True) -> None:
         super().__init__()
         self.log = log
         self.time_steps = time_steps
         self.fc1 = nn.Linear(input_size, hidden_size)
-        encoder_layers = nn.TransformerEncoderLayer(d_model=output_size, nhead=nhead, dim_feedforward=hidden_dim)
         self.fc2 = nn.Linear(hidden_size, output_size)
-        encoder_layers = nn.TransformerEncoderLayer(d_model=output_size, nhead=nhead, dim_feedforward=hidden_dim, batch_first=True)
+        encoder_layers = nn.TransformerEncoderLayer(d_model=output_size+embed_dim, nhead=nhead, dim_feedforward=hidden_dim, batch_first=True)
         self.transformer_encoder = nn.TransformerEncoder(encoder_layers, num_layers)
-        self.fc3 = nn.Linear(output_size, output_size)
+        self.fc3 = nn.Linear(output_size+embed_dim, output_size)
         self.fc4 = nn.Linear(output_size, 2)
+        self.action_embedding = nn.Linear(self.time_steps, embed_dim)
```

Comment:
I am not sure if we should collapse the time dimension here. For me it would be more intuitive, if the output is of form (T, E) where E is the embedding size. We then pass the current action encodings to the transformer at the corresponding time step